src/mpi/coll/transports/generic/transport.c

```
@@ -0,0 +1,1243 @@
+                                       (void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_L) complete\n"));
+            } else {
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Right reduction vtxp->nbargs.reduce_local.flags %ld \n",
+                         vtxp->nbargs.reduce_local.flags));
+                /* An extra data copy is required to do the R reduction */
+                MPIR_Reduce_local((const void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       (void *) vtxp->nbargs.reduce_local.inbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+
+                MPIR_Localcopy(vtxp->nbargs.reduce_local.inbuf,
+                               vtxp->nbargs.reduce_local.count,
+                               vtxp->nbargs.reduce_local.dt,
+                               vtxp->nbargs.reduce_local.inoutbuf,
+                               vtxp->nbargs.reduce_local.count, vtxp->nbargs.reduce_local.dt);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_R) complete\n"));
+            }
+
+            /* record vertex completion */
+            MPIR_COLL_GENERIC_record_vtx_completion(vtxp, sched);
+            break;
+        }
+    }
+
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIR_COLL_GENERIC_ISSUE_VTX);
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIR_COLL_GENERIC_poke
+/* Transport function to make progress on the schedule */
+int MPIR_COLL_GENERIC_poke(MPIR_COLL_GENERIC_sched_t * sched)
```

This function should return an error code and the value of whether or not the schedule is done should be an `int *` input argument.

---

Ok.

---

Done